benchmark/single-source/NSStringConversion.swift

```diff
@@ -14,16 +14,93 @@
 import TestsUtils
 import Foundation
 
-public let NSStringConversion = BenchmarkInfo(
-  name: "NSStringConversion",
-  runFunction: run_NSStringConversion,
-  tags: [.validation, .api, .String, .bridging])
+public let NSStringConversion = [
+  BenchmarkInfo(name: "NSStringConversion",
+                runFunction: run_NSStringConversion,
+                tags: [.validation, .api, .String, .bridging]),
+  BenchmarkInfo(name: "NSStringConversion.UTF8",
+                runFunction: run_NSStringConversion_nonASCII,
+                tags: [.validation, .api, .String, .bridging],
+                setUpFunction: setUpNSStringConversion_nonASCII),
+  BenchmarkInfo(name: "NSStringConversion.Mutable",
+                runFunction: run_NSMutableStringConversion,
+                tags: [.validation, .api, .String, .bridging],
+                setUpFunction: setUpNSMutableStringConversion),
+  BenchmarkInfo(name: "NSStringConversion.Long",
+                runFunction: run_NSStringConversion_long,
+                tags: [.validation, .api, .String, .bridging],
+                setUpFunction: setUpNSStringConversion_long),
+  BenchmarkInfo(name: "NSStringConversion.LongUTF8",
+                runFunction: run_NSStringConversion_longNonASCII,
+                tags: [.validation, .api, .String, .bridging],
+                setUpFunction: setUpNSStringConversion_longNonASCII)]
 
 public func run_NSStringConversion(_ N: Int) {
 #if _runtime(_ObjC)
 let test:NSString = NSString(cString: "test", encoding: String.Encoding.ascii.rawValue)!
   for _ in 1...N * 10000 {
+    //Doesn't test accessing the String contents to avoid changing historical benchmark numbers
     blackHole(identity(test) as String)
   }
 #endif
 }
+
+fileprivate var test:NSString = ""
+
+fileprivate func innerLoop(_ str: NSString, _ N: Int, _ scale: Int = 5000) {
+  for _ in 1...N * scale {
+    for char in (identity(str) as String).utf8 {
+      blackHole(char)
+    }
+  }
+}
+
+public func setUpNSStringConversion_nonASCII() {
+  #if _runtime(_ObjC)
+  test = NSString(cString: "tëst", encoding: String.Encoding.utf8.rawValue)!
+  #endif
+}
+
+public func run_NSStringConversion_nonASCII(_ N: Int) {
+  #if _runtime(_ObjC)
+  innerLoop(test, N)
+  #endif
+}
+
+public func setUpNSMutableStringConversion() {
+  #if _runtime(_ObjC)
+  test = NSMutableString(cString: "test", encoding: String.Encoding.ascii.rawValue)!
+  #endif
+}
+
+public func run_NSMutableStringConversion(_ N: Int) {
+  #if _runtime(_ObjC)
+  let test:NSString = NSMutableString(cString: "test", encoding: String.Encoding.ascii.rawValue)!
```

Comment:
This should use the `test` instance created in `setUp`, shouldn't it?

Reply:
Oops, yes. Missed that when converting!